.circleci/config.yml

```diff
@@ -69,109 +57,103 @@ jobs:
-            if ! echo "$(aws --version)" | grep -q "(aws-cli/2)"; then
-                echo "Version 2 installed successfully."
-            else
-                echo "Incorrect version installed"
-                exit 1
-            fi
 
 workflows:
-  lint-pack_validate_deploy-dev:
+  # Prior to producing a development orb (which requires credentials) basic validation, linting, and even unit testing can be performed.
+  # This workflow will run on every commit
+  test-pack:
     unless: << pipeline.parameters.run-integration-tests >>
     jobs:
-      - orb-tools/lint
-
-      - orb-tools/pack
-
+      - orb-tools/lint # Lint Yaml files
+      - orb-tools/pack # Pack orb source
+      - shellcheck/check:
+          dir: ./src/scripts
+          exclude: SC2148
+      # Publish development version(s) of the orb.
       - orb-tools/publish-dev:
-          context: orb-publishing
           orb-name: circleci/aws-cli
-          requires: [orb-tools/pack, orb-tools/lint]
-
+          context: orb-publisher # A restricted context containing your private publishing credentials. Will only execute if approved by an authorized user.
+          requires:
+            - orb-tools/lint
+            - orb-tools/pack
+            - shellcheck/check
+      # Trigger an integration workflow to test the
+      # dev:${CIRCLE_SHA1:0:7} version of your orb
       - orb-tools/trigger-integration-tests-workflow:
-          context: orb-publishing
-          requires: [orb-tools/publish-dev]
+          name: trigger-integration-dev
+          context: orb-publisher
+          requires:
+            - orb-tools/publish-dev
 
-  integration-tests_deploy-prod:
+  # This `integration-test_deploy` workflow will only run
+  # when the run-integration-tests pipeline parameter is set to true.
+  # It is meant to be triggered by the "trigger-integration-tests-workflow"
+  # job, and run tests on <your orb>@dev:${CIRCLE_SHA1:0:7}.
+  integration-test_deploy:
     when: << pipeline.parameters.run-integration-tests >>
     jobs:
-      - integration-test-install-v2:
+      # Run any integration tests defined within the `jobs` key.
+      - integration-test-install:
           matrix:
             parameters:
-              executor: [aws-cli/default, without-less, macos, machine, python, awsv2preinstalled]
-      - integration-test-install-v1:
-          matrix:
-            parameters:
-              executor: [macos, machine, python]
-      - integration-test-skip-install:
-          matrix:
-            parameters:
-              executor: [macos, machine, python]
-      - replace-pip-aws-test
-      # This job is supposed to fail, do not run it unless for testing purposes.
-      # This job fails with a more useful error message when a user wants to install CLI V1 on an image without python.
-      # - integration-test-install-v1:
-      #     name: v1-docker
-      #     executor: without-less
-
+              executor: ["docker-base", "macos", "arm"]
```

Comment:
This PR adds the Linux ARm image but drops the Linux x86_64 image. I think we should test with both.

Reply:
Added